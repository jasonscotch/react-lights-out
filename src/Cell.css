.Cell {
  height: 100px;
  width: 100px;
  background-color: rgba(0, 0, 0, 0.63);
  cursor: pointer;
  border: solid 1px black;
  border-radius: 25%;
  transition: transform 0.3s;
}

.Cell-lit {
  background-color: yellow;
}

.Cell:hover {
  transform: scale(1.1);
}

