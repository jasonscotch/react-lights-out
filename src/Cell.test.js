import { render } from '@testing-library/react';
import Cell from './Cell';
  

    let table;
    
    beforeEach(() => {
      let tr = document.createElement("tr");
      table = document.body.appendChild(tr);
    });
  
    it("renders without crashing", () => {
      render(<Cell />, { table });
    });
  
    it("matches snapshot - lit", () => {
      const { asFragment } = render(<Cell isLit />, { table });
      expect(asFragment()).toMatchSnapshot();
    });
  
    it("matches snapshot - not lit", () => {
      const { asFragment } = render(<Cell />, { table });
      expect(asFragment()).toMatchSnapshot();
    });
  