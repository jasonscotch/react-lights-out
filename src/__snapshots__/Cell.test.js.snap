// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot - lit 1`] = `<DocumentFragment />`;

exports[`matches snapshot - not lit 1`] = `<DocumentFragment />`;
